deploy/cdk/lib/foundation/foundation-stack.ts

```diff
@@ -0,0 +1,123 @@
+import { Certificate, CertificateValidation, ICertificate } from "@aws-cdk/aws-certificatemanager";
+import { IVpc, Vpc } from "@aws-cdk/aws-ec2";
+import { Cluster, ICluster } from "@aws-cdk/aws-ecs";
+import { ILogGroup, LogGroup, RetentionDays } from "@aws-cdk/aws-logs";
+import { HostedZone, IHostedZone } from "@aws-cdk/aws-route53";
+import { Bucket, BucketAccessControl, HttpMethods, IBucket } from "@aws-cdk/aws-s3";
+import { Construct, Duration, RemovalPolicy, Stack, StackProps } from "@aws-cdk/core";
+
+
+export interface IBaseStackProps extends StackProps {
+  /**
+   * The domain name to use for Route53 zones and recordsets and
+   * ACM certificates
+   */
+  readonly domainName: string;
+
+  /**
+   * Should this stack create a Route53 Zone? Default is false
+   */
+  readonly doCreateZone?: boolean;
+
+  /**
+   * If this stack import from an existing Route53 Zone, provide the zone id
+   */
+  readonly useDnsZone?: string;
+}
+
+export class FoundationStack extends Stack {
+  /**
+   * The VPC to place all services related to this application
+   */
+  public readonly vpc: IVpc;
+
+  /**
+   * The Route53 zone (only created if doCreateZone is true)
+   */
+  public readonly hostedZone: IHostedZone;
+
+  /**
+   * Wildcard certificate for all components of this application
+   */
+  public readonly certificate: ICertificate;
+
+  /** 
+   * Shared cluster for any ECS tasks/services in this application
+   */
+  public readonly cluster: ICluster;
```

Comment:
What containerized services are we running anymore? If none, we should remove.

Reply:
Manifest pipeline still uses ECS tasks.

Reply:
Alright. I was looking in libnd and didn't see anything running, but I remember now that manifest uses Fargate as a task runner, right?